### DISCUSSION TITLE
Another remix i18n lib: remix-intl

### DISCUSSION THREAD

**Discussion by suhaotian** [remix-run/remix#9188] [Show and tell / tips]
Hey, guys,

I just made another i18n lib for Remix [remix-intl](https://github.com/suhaotian/remix-intl)

Currently early stage, just worked, If you have any questions please tell me!

 Compared to `remix-i18next`:

- Both use `i18next`
- In `remix-intl`, no need fetch i18n data before hydrate in client side, already send data by a trick  in `<IntlScript />`
- You fully control dynamic load `locales` or `messages` from another API or in public folder
- Another choice, why not?

**Comment by msurdi** [comment]
Nice work!

To be honest, I'm generally put off by these solutions because of how complex the setup is and the many dependencies and moving parts involved. So far I've been doing [just this](https://github.com/msurdi/remix-i18n-example) and it's been working fine for me.

**Reply by suhaotian** [reply]
Thanks sharing and feedback! Your example code looks ok, but in my case, what if the languages and i18n keys data are dynamic from API?

**Reply by msurdi** [reply]
Do you mean like having to retrieve those from a CMS "on the fly"? I never had such an use case. Thinking about it I might just fetch them once and cache them server side, and then keep this current approach. But that highly depends on the specifics of the use case I guess.

**Reply by suhaotian** [reply]
Yeah `on the fly`, from seperate API, someone in the team maintain the i18n data
